File: k-kbk/4주차/8979.py
n, k = map(int, input().split())

countries = []
target = []

for _ in range(n):
    temp = list(map(int, input().split()))
    countries.append(temp)

    if temp[0] == k:
        target = temp[1:]

countries.sort(key=lambda x: (-x[1], -x[2], -x[3]))

for i in range(n):
    if countries[i][1:] == target:
        print(i + 1)
        break

File: keemsebin/4주차/8979.js
const readline = require("readline");
const rl = readline.createInterface({
  input: process.stdin,
  output: process.stdout,
});

let input = [];

rl.on("line", function (line) {
  input.push(line);
}).on("close", function () {
  let [N, K] = input[0].split(" ").map(Number);

  let score = [];

  for (let i = 1; i <= N; i++) {
    let [num, a, b, c] = input[i].split(" ").map(Number);
    score.push({ num, a, b, c });
  }

  score.sort((a, b) => {
    if (a.a !== b.a) {
      return b.a - a.a;
    }
    if (a.b !== b.b) {
      return b.b - a.b;
    }
    return b.c - a.c;
  });

  let rank = 1;

  for (let i = 0; i < N; i++) {
    if (score[i].num === K) {
      console.log(rank);
      break;
    }
    if (
      i < N - 1 &&
      score[i].a === score[i + 1].a &&
      score[i].b === score[i + 1].b &&
      score[i].c === score[i + 1].c
    ) {
      continue;
    }
    rank++;
  }
});
